onnxruntime/core/providers/webgpu/webgpu_external_header.h

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
+
+#include <webgpu/webgpu_cpp.h>
```

Comment:
This change is for suppressing strict-aliasing warnings from the 3rd-party webgpu source code, not ours.